include/nlohmann/ordered_map.hpp

```diff
@@ -43,17 +42,24 @@ template <class Key, class T, class IgnoredLess = std::less<Key>,
 
     // Explicit constructors instead of `using Container::Container`
     // otherwise older compilers choke on it (GCC <= 5.5, xcode <= 9.4)
-    ordered_map() noexcept(noexcept(Container())) : Container{} {}
-    explicit ordered_map(const Allocator& alloc) noexcept(noexcept(Container(alloc))) : Container{alloc} {}
-    template <class It>
+    ordered_map() noexcept(noexcept(Container()))
+        : Container{}
+    {}
+    explicit ordered_map(const Allocator& alloc) noexcept(noexcept(Container(alloc)))
+        : Container{alloc}
+    {}
+    template<class It>
     ordered_map(It first, It last, const Allocator& alloc = Allocator())
-        : Container{first, last, alloc} {}
-    ordered_map(std::initializer_list<value_type> init, const Allocator& alloc = Allocator() )
-        : Container{init, alloc} {}
+        : Container{first, last, alloc}
+    {}
+    ordered_map(std::initializer_list<value_type> init, const Allocator& alloc = Allocator())
+        : Container{init, alloc}
+    {}
 
     std::pair<iterator, bool> emplace(const key_type& key, T&& t)
     {
-        for (auto it = this->begin(); it != this->end(); ++it)
+        auto endItr = this->end();
```

Comment:
The [Container requirements](https://en.cppreference.com/w/cpp/named_req/Container) require `end()` to have constant complexity. I wonder whether this change would actually create different assembler code. Did you profile this?

Reply:
I did not profile this.
The idea was to save the value in a variable instead of calling the same function several times

Reply:
Yes, but the `end()` function is cheap. Furthermore, the change could change the semantics of the code if the range is changed during iteration (which it is not here, but in other functions.